src/software/sensor_fusion/sensor_fusion.cpp

```diff
@@ -198,17 +196,22 @@ bool SensorFusion::shouldTrustRobotStatus()
         return false;
     }
 
-    double distance =
-        (robot_with_ball_in_dribbler.value().position() - ball.value().position())
-            .length();
+    // Check if vision detects a ball on the field
+    if (!ball.has_value())
+    {
+        return false;
+    }
```

Comment:
Shouldn't we trust robot status if ball isn't visible by vision?

Reply:
I think the comment may be misleading. 

We need ball because we need to compute a distance between the breakbeam and the robot.

Reply:
Actually, that also make sense. 

Used to be that we check for all the optional value, and if we can't unwrap all the optional value, we would false.